### ISSUE TITLE
stripe-android not compatible with Flutter 2.12.0-4.1.pre (beta)

### ISSUE THREAD

**Issue by nerder** [flutter-stripe/flutter_stripe#652]
**Describe the bug**

By upgrading flutter being in the `beta` channel is not possible to complie the project anymore.

**To Reproduce**
1. flutter create
2. flutter channel beta
3. flutter run

**Expected behavior**

To being able to compile

**Additional context**

Installed version of Stripe: `flutter_stripe: ^2.4.0`

```
➜ flutter doctor -v
[✓] Flutter (Channel beta, 2.12.0-4.1.pre, on macOS 11.6.2 20G314 darwin-x64, locale en-ES)
    • Flutter version 2.12.0-4.1.pre at /Users/nerder/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 680962aa75 (7 days ago), 2022-03-16 13:46:03 -0700
    • Engine revision e9f57b5d0f
    • Dart version 2.17.0 (build 2.17.0-182.1.beta)
    • DevTools version 2.11.1

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/nerder/Library/Android/sdk
    • Platform android-31, build-tools 30.0.2
    • ANDROID_SDK_ROOT = /Users/nerder/Library/Android/sdk
    • Java binary at: /Users/nerder/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7935034/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/nerder/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7935034/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/nerder/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3)
    • IntelliJ at /Users/nerder/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 63.2.4
    • Dart plugin version 213.5744.122

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3)
    • IntelliJ at /Users/nerder/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/213.5744.223/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2021.2.3)
    • IntelliJ at /Users/nerder/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/212.5457.46/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] Connected device (3 available)
    • SM G975F (mobile) • R58M67B0S3Y • android-arm64  • Android 12 (API 31)
    • macOS (desktop)   • macos       • darwin-x64     • macOS 11.6.2 20G314 darwin-x64
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 99.0.4844.83

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Comment by nerder** [comment]
After a bit of investigation the issue seems to be related to a breaking change made by Flutter, by changing the signature of the method `PlatformViewFactory::create`

Original code (not compiling):
```kotlin
class StripeSdkGooglePayButtonPlatformViewFactory(
        private val flutterPluginBinding: FlutterPlugin.FlutterPluginBinding,
        private val payButtonManager: GooglePayButtonManager,
        private val sdkAccessor: () -> StripeSdkModule
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val channel = MethodChannel(flutterPluginBinding.binaryMessenger, "flutter.stripe/google_pay_button/${viewId}")
        val creationParams = args as? Map<String?, Any?>?
        return StripeSdkGooglePayButtonPlatformView(context, channel, viewId, creationParams, payButtonManager, sdkAccessor)
    }
}
```

Possible fix:

```kotlin
class StripeSdkGooglePayButtonPlatformViewFactory(
        private val flutterPluginBinding: FlutterPlugin.FlutterPluginBinding,
        private val payButtonManager: GooglePayButtonManager,
        private val sdkAccessor: () -> StripeSdkModule
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context?, viewId: Int, args: Any?): PlatformView {
        val channel = MethodChannel(flutterPluginBinding.binaryMessenger, "flutter.stripe/google_pay_button/${viewId}")
        val creationParams = args as? Map<String?, Any?>?
        return StripeSdkGooglePayButtonPlatformView(context!!, channel, viewId, creationParams, payButtonManager, sdkAccessor)
    }
}
```

Now while i'm not sure if this is a possible valid fix, I can see that the `context` is never accessed so skipping the null-check might be a valid alternative to comply with the new signature. Of course this will not be compatible with both `stable` and `beta` at this time, so a branch from where people can fetch `flutter_stripe` is required.

Any help?

**Comment by remonh87** [comment]
@nerder  thanks for the thorough investigation.  We will see if we can make a fix that don't break the original Flutter stable implementation. Else we will wait with adding the new code when google integrated the code in the stable channel.

**Comment by nerder** [comment]
Ok, maybe i'll prepare a fix for beta in a PR, so people using `beta` can source from that commit. I don't really think there is a possible backward compatible fix here.

**Comment by kamami** [comment]
Still an issue on beta. Unfortunately I can not use stable, because of another conflict (DraggableScrollableController is not a listenable anymore on stable).

**Comment by kamami** [comment]
@nerder When do you plan to publish your fix to beta?

**Comment by nerder** [comment]
> @nerder thanks for the thorough investigation. We will see if we can make a fix that don't break the original Flutter stable implementation. Else we will wait with adding the new code when google integrated the code in the stable channel.

I don't really think this is one me right now, sadly.

**Comment by kamami** [comment]
Puh ok, thanks for the response. Seems like we have to be patient, even if it is really critical and urgent.

**Comment by jonasbark** [comment]
If we merge and release it then incompatibilities with Flutter stable can emerge and that would be much worse. If you need it right now then all you need to do is to refer to the sources from this PR https://github.com/flutter-stripe/flutter_stripe/pull/654

**Comment by kamami** [comment]
What do you mean by "refer to the sources"? 

I tried to get the package via:

```
flutter_stripe:
    git:
     url: https://github.com/deathcoder/flutter_stripe.git
     ref: main
```

But this resolves in an error:
`pub get failed (1; Could not find a file named "pubspec.yaml" in https://github.com/deathcoder/flutter_stripe.git e371670ebab0df1991e03d93263b06d2a272df08.)`

**Comment by jonasbark** [comment]
You'll have to add the path to refer to the packages itself. 
Also dependency_overrides are necessary, similar to here:
https://github.com/flutter-stripe/flutter_stripe/blob/main/example/pubspec.yaml#L26

Read more about it here:
https://dart.dev/tools/pub/dependencies

**Comment by kamami** [comment]
I did manage to solve it btw. it was the second option with `path packages` and the `dependency_overrides`.

**Comment by nerder** [comment]
@jonasbark is this in stable right now with the recent Flutter v3?

**Comment by remonh87** [comment]
Fixed in release v3.0 of our library
